salt/minion.py

```diff
@@ -585,9 +585,10 @@ def __init__(self, opts):
         # Clean out the proc directory (default /var/cache/salt/minion/proc)
         if (self.opts.get('file_client', 'remote') == 'remote'
                 or self.opts.get('use_master_when_local', False)):
-            if HAS_ZMQ:
-                zmq.eventloop.ioloop.install()
-            io_loop = LOOP_CLASS.current()
+            if self.opts['transport'] == 'zeromq' and HAS_ZMQ:
+                io_loop = zmq.eventloop.ioloop.ZMQIOLoop()
+            else:
+                io_loop = LOOP_CLASS.current()
```

Comment:
Actually, I just realized what I said isn't 100% true. The first clause of the "if" case doesn't call `current()`, but the second clause does. So the first clause will create a new IOLoop even if one already exists, and the second will create a new IOLoop only if one doesn't already exist, and use the existing one if it already exists.

My mistake.